lib/spack/spack/build_environment.py

```diff
@@ -452,6 +451,17 @@ def set_build_environment_variables(pkg, env, dirty):
     return env
 
 
+def determine_number_of_jobs(
+        cap,
+        command_line=spack.config.get('config:build_jobs', scope='command_line'),
+        default=spack.config.get('config:build_jobs')):
+    return command_line if command_line is not None else min(default, cap)
+
+
+def determine_number_of_jobs_for_package(pkg, cpus=cpus_available()):
```

Comment:
Don't run a method inside the function signature.

This case is less certain to cause a problem, but it's still not a good idea. The default gets evaluated at function definition time, and changes over time can lead to truly awful bugs.

Reply:
Right, I have been spoiled with programming in julia, didn't realize python worked differently.

Reply:
This is fixed